3.6/debian-9/rootfs/opt/bitnami/scripts/libmongodb.sh

```diff
@@ -503,14 +588,45 @@ EOF
         )
     fi
 
-    if [[ -n "$MONGODB_USERNAME" ]] && [[ -n "$MONGODB_PASSWORD" ]] && [[ -n "$MONGODB_DATABASE" ]]; then
-        info "Creating '$MONGODB_USERNAME' user..."
-
-        result=$(
-            mongodb_execute 'root' "$MONGODB_ROOT_PASSWORD" "" "127.0.0.1" <<EOF
-db.getSiblingDB('$MONGODB_DATABASE').createUser({ user: '$MONGODB_USERNAME', pwd: '$MONGODB_PASSWORD', roles: [{role: 'readWrite', db: '$MONGODB_DATABASE'}] })
+    if [[ -n "$MONGODB_USERNAMES" ]]; then
+        # Fill in arrays called databases, usernames and passwords with
+        # information from matching environment variables.
+        mongodb_auth
+
+        if [[ -n "$MONGODB_DATABASES" ]]; then
+            # Loop over the databases, usernames and passwords arrays, creating
+            # each user in the database at the same index.
+            for (( i=0; i<${#databases[@]}; i++ )); do
+                # MongoDB refuses to create users without passwords. It would
+                # return the following otherwise:
+                # Error: couldn't add user: User passwords must not be empty
+                if [[ -n "${passwords[i]}" ]]; then
+                    info "Creating user '${usernames[i]}' in database '${databases[i]}'..."
+
+                    result=$(
+                        mongodb_execute 'root' "$MONGODB_ROOT_PASSWORD" "" "127.0.0.1" <<EOF
+db.getSiblingDB('${databases[i]}').createUser({ user: '${usernames[i]}', pwd: '${passwords[i]}', roles: [{role: 'readWrite', db: '${databases[i]}'}] })
 EOF
-        )
+                    )
+                fi
+            done
+        else
+            # Loop over all users and create them within the default database.
+            for (( i=0; i<${#usernames[@]}; i++ )); do
+                # MongoDB refuses to create users without passwords. It would
+                # return the following otherwise:
+                # Error: couldn't add user: User passwords must not be empty
+                if [[ -n "${passwords[i]}" ]]; then
+                    info "Creating user '${usernames[i]}' in default database..."
+
+                    result=$(
+                        mongodb_execute 'root' "$MONGODB_ROOT_PASSWORD" "" "127.0.0.1" <<EOF
+db.getSiblingDB(db.stats().db).createUser({ user: '${usernames[i]}', pwd: '${passwords[i]}', roles: [{role: 'readWrite', db: db.getSiblingDB(db.stats().db).stats().db }] })
```

Comment:
Wouldn't this also work?
```suggestion
db.getSiblingDB(db.stats().db).createUser({ user: '${usernames[i]}', pwd: '${passwords[i]}', roles: [{role: 'readWrite', db: db.stats().db }] })
```

Reply:
I ... don't know. All the code seems to be accessing via `getSiblingDB()`, I simply do not know why. Your proposal seems to work, but maybe are there edge cases?